packages/pigeon/lib/java_generator.dart

```diff
@@ -218,38 +202,58 @@ Result<$returnType> $resultName = new Result<$returnType>() {
-                } else if (method.returnType.isVoid) {
-                  indent.writeln('$call;');
-                  indent.writeln('wrapped.put("${Keys.result}", null);');
-                } else {
-                  indent.writeln('$returnType output = $call;');
-                  indent.writeln('wrapped.put("${Keys.result}", output);');
-                }
-              });
-              indent.write('catch (Error | RuntimeException exception) ');
-              indent.scoped('{', '}', () {
-                indent.writeln(
-                    'wrapped.put("${Keys.error}", wrapError(exception));');
-                if (method.isAsynchronous) {
-                  indent.writeln('reply.reply(wrapped);');
-                }
-              });
-              if (!method.isAsynchronous) {
-                indent.writeln('reply.reply(wrapped);');
-              }
-            });
+              methodArgument.add(resultName);
+            }
+            final String call =
+                'api.${method.name}(${methodArgument.join(', ')})';
+            if (method.isAsynchronous) {
+              indent.writeln('$call;');
+            } else if (method.returnType.isVoid) {
+              indent.writeln('$call;');
+              indent.writeln('wrapped.put("${Keys.result}", null);');
+            } else {
+              indent.writeln('$returnType output = $call;');
+              indent.writeln('wrapped.put("${Keys.result}", output);');
+            }
           });
-          indent.scoped(null, '}', () {
-            indent.writeln('channel.setMessageHandler(null);');
+          indent.write('catch (Error | RuntimeException exception) ');
+          indent.scoped('{', '}', () {
+            indent
+                .writeln('wrapped.put("${Keys.error}", wrapError(exception));');
+            if (method.isAsynchronous) {
+              indent.writeln('reply.reply(wrapped);');
+            }
           });
+          if (!method.isAsynchronous) {
+            indent.writeln('reply.reply(wrapped);');
+          }
         });
-      }
+      });
+      indent.scoped(null, '}', () {
+        indent.writeln('channel.setMessageHandler(null);');
+      });
+    });
+  }
+
+  indent.writeln(
+      '/** Generated interface from Pigeon that represents a handler of messages from Flutter.*/');
+  indent.write('public interface ${api.name} ');
+  indent.scoped('{', '}', () {
+    api.methods.forEach(writeInterfaceMethod);
+    indent.addln('');
+    final String codecName = _getCodecName(api);
+    indent.format('''
+/** The codec used by ${api.name}. */
+static MessageCodec<Object> getCodec() {
+\treturn $codecName.INSTANCE;
+}
+''');
+    indent.writeln(
+        '/** Sets up an instance of `${api.name}` to handle messages through the `binaryMessenger`. */');
+    indent.write(
+        'static void setup(BinaryMessenger binaryMessenger, ${api.name} api) ');
+    indent.scoped('{', '}', () {
+      api.methods.forEach(writeMethodSetup);
```

Comment:
For future reference: it would be helpful to separate a refactor that makes ~the entire function into a diff from functional changes. I have no idea what actually changed in this code beyond the indentation changes from the inner function extraction.